PJ05/src/MessageSystem.java

```diff
@@ -27,17 +26,17 @@ public User getUser(String name) throws UserNotFoundException {
                 return user;
             }
         }
+
+
         throw new UserNotFoundException();
     }
 
     public User getUser(UUID find_uuid) throws UserNotFoundException {
-        for (UUID uuid : userDatabase.uuids()) {
-            User user = userDatabase.get(uuid);
-            if (user.uuid.equals(find_uuid)) {
-                return user;
-            }
-        }
-        throw new UserNotFoundException();
+
+
+        User user = userDatabase.get(find_uuid);
```

Comment:
You need to catch an exception and rethrow here. `Database.get()` will probably only throw generic exceptions like `KeyNotFoundError`. Similar for `getConversation` and `getMessage`.